File: 백준/Silver/11047. 동전 0/동전 0.py
N, K = map(int, input().split())
coin = []
for i in range(N):
    coin.append(int(input()))

answer = 0

for i in reversed(coin):
    if K >= i:
        answer += K // i
        K = K % i
    elif K == 0:
        break
    else:
        continue

print(answer)
    

File: 백준/Silver/11047. 동전 0/동전 0.java
import java.util.*;
public class Main{
    public static void main(String[]args){
        
        Scanner in = new Scanner(System.in);
        int N = in.nextInt();
        int K = in.nextInt();
        int [] coin = new int[N];
        int answer = 0;
        for(int i =0; i<N; i++){
            coin[i] = in.nextInt();
        }
        
        for(int i = N-1; i>=0; i--){
            if(K >= coin[i]){
                answer += K/coin[i];
                K %= coin[i];
            }
        }
        
        System.out.print(answer); 
    }
}